hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/hudi/command/CreateHoodieTableCommand.scala

```diff
@@ -164,6 +161,20 @@ object CreateHoodieTableCommand {
     }
   }
 
+  private def convertFieldTypes(fieldTypes: StructType): StructType = {
+    val fields = new Array[StructField](fieldTypes.fields.length)
+    var index = 0
```

Comment:
Should we just fix the hive sync schema instead? Modification in here affects beyond the Hive scope I guess.